pytorch_lightning/loggers/neptune.py

```diff
@@ -523,6 +523,16 @@ def after_save_checkpoint(self, checkpoint_callback: "ReferenceType[ModelCheckpo
             file_names.add(model_name)
             self.experiment[f"{checkpoints_namespace}/{model_name}"].upload(key)
 
+        # log best model path and checkpoint
+        if checkpoint_callback.best_model_path:
+            self.experiment[
+                self._construct_path_with_prefix("model/best_model_path")
+            ] = checkpoint_callback.best_model_path
```

Comment:
quick ques: why aren't we following the similar upload process as that of last model path?
for eg:
we are not doing this while uploading last model:
```py
self._construct_path_with_prefix("model/best_model_path")
```

Reply:
@Raalsky could you have a look?

Reply:
@kamil-kaczmarek Would you mind? :wink:

Reply:
Sorry that it takes so long to answer :relieved:.

The main idea was to provide an exact local path of the corresponding model checkpoint files that were uploaded. At our design level, there was some lack of information about how we should fully handle every behavior that ModelCheckpoint could provide. After that, at the implementation, we missed and messed something that it's got only applied to the best model.

I think it should be translated to something like `local_path` instead of `best_model_path` and be prepared for every last/k-best/best (especially be exactly right to every `cpkt` file). We would like to do some additional discussions to take proper care of it and I think it's out-of-scope for this PR. It may introduce new functionality or change the behavior instead of only bug-fixing.